src/main/java/org/spongepowered/api/world/extent/Extent.java

```diff
@@ -131,6 +132,28 @@ default Vector3i getHighestPositionAt(Vector3i position) {
     }
 
     /**
+     * Returns the y level that precipitation ends falling in the given column.
+     *
+     * <p>A value is still returned for columns in biomes which do not
+     * receive precipitation.</p>
+     *
+     * @return The y level that precipitation ends
+     */
+    int getPrecipitationLevel(int x, int z);
+
+    /**
+     * Returns the y level that precipitation ends falling in the given column.
+     *
+     * <p>A value is still returned for columns in biomes which do not
+     * receive precipitation.</p>
+     *
+     * @return The y level that precipitation ends
+     */
+    default int getPrecipitationLevel(Vector2i column) {
```

Comment:
[The same could be said of the methods I referenced](https://github.com/Desetude/SpongeAPI/blob/16c731482f87594ec4f5ef9b0fd654a51da2fc31/src/main/java/org/spongepowered/api/world/extent/Extent.java#L109-L132) - they aren't really anything to do with 3d locations, but it has a `Vector3i` method for sake of ease. Most developers will have a `Vector3i` to hand, rather than a `Vector2i` when wanting this location. It's also useful to return a `Vector3i` so a developer can use the position immediately without having to construct the vector themselves.

I don't disagree with your point in that the y co-ordinate is irrelavent, certainly, but I'd like to see the method signatures be consistent with one another.